docs/usage/key-concepts/changelogs.md

```diff
@@ -0,0 +1,109 @@
+
+For more details, see [Private packages, looking up changelogs](../getting-started/private-packages.md#looking-up-changelogs).
+
+## Relevant configuration options
+
+### [`fetchChangelogs`](../configuration-options.md#fetchchangelogs)
+
+Set to `off` if changelog fetching is causing a problem.
+
+Set to `branch` if you have an advanced use case where you're embedding changelogs in the Git commit itself, we don't recommend this due to its potential size.
+
+### [`customChangelogUrl`](../configuration-options.md#customchangelogurl)
+
+This doesn't help with _fetching_ the changelogs, but if you configure it then Renovate will include a link to this URL in the PR body, so users can click through to read the changelog.
+
+## Platforms that Renovate can fetch changelogs from
+
+See the list of platforms in the [`fetchChangelogs` config option docs](../configuration-options.md#fetchchangelogs).
+
+### Running Renovate on a non-GitHub platform
+
+Most Open Source packages are hosted on github.com, which means most changelogs are hosted there too.
+Fetching changelogs from github.com requires a GitHub token because GitHub blocks unauthenticated GraphQL API use.
+
+This means that if you run Renovate on self-hosted GitHub Enterprise Server, or any non-GitHub platform which Renovate supports, then you need to configure a github.com Personal Access Token in Renovate in order to fetch changelogs.
+
+Read [Running Renovate, GitHub.com token for changelogs](../getting-started/running.md#githubcom-token-for-changelogs) to learn more.
+
+## Troubleshooting missing changelogs
+
+Follow these steps to find out why Renovate does not find a changelog:
+
+1. The datasource for this package does not support sourceUrls.
+   - If the registry fundamentally does not provide this data, then the only possibility is for it to be manually populated through PRs to Renovate's source code
+   - If the registry provides source URLs but Renovate does not support it yet, then raise a feature request with examples, or better yet a Pull Request to implement it yourself
+   - Sometimes self-hosted versions of registries don't include the full metadata compared to what the public registries do
+1. The package does not provide this information to the registry.
+   - For example, occasionally `npm` packages don't have `repository` fields included
+   - For example, Docker images regularly do not have the required `LABEL` entry
+1. Renovate cannot access the source repository
+   - This is typically a concern for private repositories only
+   - Check if the token Renovate uses has access rights to the repository you need it to access
+1. Renovate cannot detect the file names or release name convention within the repository
+   - In this case an enhancement to Renovate might be needed to better detect the releases/formats, assuming it's got a reasonable convention to follow
```

Comment:
Same question about `assuming it's got a reasonable convention to follow`. I guess you really mean something like this?

> assuming the upstream package/ecosystem has a reasonable convention that Renovate can follow